mlrun/api/utils/clients/chief.py

```diff
@@ -206,59 +211,84 @@ def _resolve_request_kwargs_from_request(
-        if not response.ok:
-            log_kwargs = copy.deepcopy(kwargs)
-            log_kwargs.update({"method": method, "path": path})
-            if response.content:
-                try:
-                    data = response.json()
-                    error = data.get("error")
-                    error_stack_trace = data.get("errorStackTrace")
-                except Exception:
-                    pass
-                else:
-                    log_kwargs.update(
-                        {"error": error, "error_stack_trace": error_stack_trace}
-                    )
-            logger.warning("Request to chief failed", **log_kwargs)
-            if raise_on_failure:
-                mlrun.errors.raise_for_status(response)
-            return response
-        # there are some responses like NO-CONTENT which doesn't return a json body
+        response = None
         try:
-            data = response.json()
-        except Exception:
-            data = response.text
-        logger.debug(
-            "Request to chief succeeded",
-            method=method,
-            url=url,
-            **kwargs,
-            response=data,
-        )
-        return response
+            response = await self._session.request(
+                method, url, verify_ssl=False, **kwargs
+            )
+            if not response.ok:
+                await self._on_request_api_failure(
+                    method, path, response, raise_on_failure, **kwargs
+                )
+            else:
+                logger.debug(
+                    "Request to chief succeeded",
+                    method=method,
+                    url=url,
+                    **kwargs,
+                )
+            yield response
+        finally:
+            if response:
+                response.release()
+
+    async def _ensure_session(self):
+        if not self._session:
+            self._session = mlrun.utils.AsyncClientWithRetry(
+                # This client handles forwarding requests from worker to chief.
+                # if we receive 5XX error, the code will be returned to the client.
+                #  if client is the SDK - it will handle and retry the request itself, upon its own retry policy
+                #  if the client is UI  - it will propagate the error to the user.
+                # Thus, do not retry.
+                # only exceptions (e.g.: connection initiating).
+                raise_for_status=False,
+            )
+
+            # if we go any HTTP response, return it, do not retry.
+            # by returning `True`, we tell the client the response is "legit" and so, it returns it to its callee.
+            self._session.retry_options.evaluate_response_callback = lambda _: True
+
+    async def _on_request_api_failure(
+        self, method, path, response, raise_on_failure, **kwargs
+    ):
+        log_kwargs = copy.deepcopy(kwargs)
+        log_kwargs.update({"method": method, "path": path})
```

Comment:
seems possible that somehow the user will pass `method/path` as one of its arguments, I would change it to have more specific name which has a lower chance to conflict with user kwargs

Reply:
that is .. aburd. it means the user will perform something like
```
client.request("POST", "/projects", method="GET", path="/functions")
```
does'nt make sense

Reply:
I remembered we has endpoints which had `path` as a query param, now looking at it, it actually was `@router.post("/projects/{project}/artifacts/{uid}/{key:path}")` which only notes to the fastapi that the last param is a path and it should not separate it .